### PR TITLE
LA-534 Create Upload Request (Simple)

### PR DESCRIPTION
**Pull request by huynguyennovem**
https://user-images.githubusercontent.com/29337364/126636212-e6c39347-3cbf-419a-b782-33bbb0706573.mov

### CODE REVIEW

File: data/lib/src/datasource_impl/upload_request_group_datasource_impl.dart [hunk 1]
```diff
@@ -59,4 +59,20 @@ class UploadRequestGroupDataSourceImpl implements UploadRequestGroupDataSource {
...
+
+  @override
+  Future<UploadRequestGroup> addNewUploadRequest(UploadRequestCreationType creationType, AddUploadRequest addUploadRequest) async {
+    return Future.sync(() async {
+      final uploadRequestGroupResponse = await _linShareHttpClient.addNewUploadRequest(creationType, addUploadRequest);
+      return uploadRequestGroupResponse.toUploadRequestGroup();
+    }).catchError((error) {
+      _remoteExceptionThrower.throwRemoteException(error, handler: (DioError error) {
+        if (error.response?.statusCode == 404) {
+          throw UploadRequestGroupsNotFound();
```
Comment (hoangdat) [inline data/lib/src/datasource_impl/upload_request_group_datasource_impl.dart:71]: it really response this error

File: data/lib/src/network/linshare_http_client.dart [hunk 2]
```diff
@@ -546,4 +547,14 @@ class LinShareHttpClient {
 
     return resultJson.map((data) => UploadRequestGroupResponse.fromJson(data)).toList();
   }
+
+  Future<UploadRequestGroupResponse> addNewUploadRequest(UploadRequestCreationType creationType, AddUploadRequest addUploadRequest) async {
+    final resultJson = await _dioClient.put(
```
Comment (hoangdat) [inline data/lib/src/network/linshare_http_client.dart:552]: IMO, it should be `POST` method

File: lib/presentation/util/extensions/datetime_extension.dart [hunk 3]
```diff
@@ -35,4 +35,9 @@ import 'package:intl/intl.dart';
 extension DateTimeExtension on DateTime {
 
   String getMMMddyyyyFormatString() => DateFormat.yMMMd().format(this);
+
+  String get_yMMMMd_formatWithJm() => DateFormat.yMMMMd().add_jm().format(this);
```
Comment (hoangdat) [inline lib/presentation/util/extensions/datetime_extension.dart:39]: What is `jm`? Besides that, IMO, you should not use this format for your method name?

**Review by hoangdat — COMMENTED** [review]
Too many things in one commit, please try to split as small as possible next time. We can split the presentation commit into some components:
- add users
- input messages
- other data
- ...
Very hard to get your idea in your solution.

File: data/lib/src/network/model/request/add_upload_body_request.dart [hunk 4]
```diff
@@ -75,7 +77,24 @@ class AddUploadBodyRequest with EquatableMixin {
 
   factory AddUploadBodyRequest.fromJson(Map<String, dynamic> json) => _$AddUploadBodyRequestFromJson(json);
 
-  Map<String, dynamic> toJson() => _$AddUploadBodyRequestToJson(this);
```
Comment (hoangdat) [inline data/lib/src/network/model/request/add_upload_body_request.dart]: What problem with it?

File: lib/presentation/util/extensions/datetime_extension.dart (hunk 3 above)
Comment (huynguyennovem) [reply lib/presentation/util/extensions/datetime_extension.dart:39]: `add_jm()` used for time format

File: data/lib/src/network/model/request/add_upload_body_request.dart (hunk 4 above)
Comment (huynguyennovem) [reply data/lib/src/network/model/request/add_upload_body_request.dart]: The generation code causes an error when parsing the request object

File: lib/presentation/util/extensions/datetime_extension.dart (hunk 3 above)
Comment (hoangdat) [reply lib/presentation/util/extensions/datetime_extension.dart:39]: It is not answered my question? What is it? Why we need it?

Comment (huynguyennovem) [reply lib/presentation/util/extensions/datetime_extension.dart:39]: It a function for time format. Example: `DateFormat.yMd().add_jm() -> 7/10/1996 5:08 PM`

Comment (hoangdat) [reply lib/presentation/util/extensions/datetime_extension.dart]: No, we should not use this schema.

**Review by hoangdat — APPROVED** [review]

**Pull request merged**